icu4c/source/common/locid.cpp

```diff
@@ -477,6 +483,42 @@ Locale &Locale::operator=(const Locale &other)
     return *this;
 }
 
+Locale& Locale::operator=(Locale&& other) U_NOEXCEPT {
+    if (this == &other) {
```

Comment:
In UnicodeString::moveFrom() I wrote
  // No explicit check for self move assignment, consistent with standard library.
  // Self move assignment causes no crash nor leak but might make the object bogus.
why differ here?

Reply:
copy assign vs. move assign are different. They should each be consistent with what's customary for each, not with each other. I spent a fair bit of time looking at what the C++ standard said about "move" and what the STL objects do. They don't check this==&other. And they don't invalidate other if they don't have to.

Reply:
Done.